### DISCUSSION TITLE
Android devices

### DISCUSSION THREAD

**Discussion by robin-gustafsson** [covidpass-org/covidpass#131] [Q&A]
How does it work for Android devices? Is there any similar certificate as Apple required?

**Comment by hauketoenjes** [comment]
Actually, there is an official way of storing "COVID vaccination and test cards" in Google Pay. But unfortunately, "you must be a healthcare organization, government agency, or an organization authorized by public health authorities to distribute COVID vaccines or tests".

See the [official documentation](https://developers.google.com/pay/passes/guides/pass-verticals/use-cases?vertical=covid-cards#pass-vertical) for more information on who is eligible to issue these certificates.

We could maybe try to ask Google for permission because this project has grown pretty big, but currently the only way is to use an app on Android that can read pkpass files. Like [Wallet Passes](https://play.google.com/store/apps/details/WalletPasses_Passbook_Wallet?id=io.walletpasses.android). Another way is to use a converter to import a pkpass in Google Pay. Like [Pass2Pay](https://play.google.com/store/apps/details?id=color.dev.com.tangerine). 

But neither of these solutions are officially supported by Android / Google.

**Reply by hauketoenjes** [reply]
Another way would be to just use the "event ticket template" for Google Pay and add the QR Code to that. But I'm not really sure if Google likes that if they have a specific type for COVID Certificates (which is understandable).

**Reply by marvinsxtr** [reply]
We are doing the same with Apple passes atm, they have a dedicated pass type for COVID19 but we just use the general pass type. There is a [table](https://developers.google.com/pay/passes/guides/introduction/about-google-pay-api-for-passes#engage-google-pay) which summarizes the capabilities of the different pass types for Google Pay. So far I don't really see major differences apart from the expiration notification for offers and the different layout for boarding passes.

**Reply by marvinsxtr** [reply]
The official documentation of the API says

> When a user clicks the button, a JSON Web Token (JWT) that represents their Pass is sent to Google servers.

This gives Google (potentially) access to all uploaded certificates.

**Comment by marvinsxtr** [comment]
Also see #99 #45
